### ISSUE TITLE
 com.docker.vpnkit is not running while on WIFI

### ISSUE THREAD

**Issue by dkd-dobberkau** [docker/for-mac#2485]
I am observing this:

Docker for Mac: version: 17.12.0-ce-mac49 (d1778b704353fa5b79142a2055a2c11c8b48a653)
macOS: version 10.12.6 (build: 16G1114)
logs: /tmp/738DC8D1-497B-41A9-854F-B472195DABAE/20180122-215312.tar.gz
failure: com.docker.vpnkit is not running
[OK]     db.git
[OK]     vmnetd
[OK]     dns
[ERROR]  driver.amd64-linux
         com.docker.driver.amd64-linux is not running
[OK]     virtualization VT-X
[OK]     app
[OK]     moby
[OK]     system
[OK]     moby-syslog
[OK]     kubernetes
[OK]     env
[OK]     virtualization kern.hv_support
[ERROR]  slirp
         Unexpected error ((Failure
  "Error connecting socket to 9p endpoint unix:/Users/olivier/Library/Containers/com.docker.docker/Data/s51: Unix.Unix_error(Unix.ECONNREFUSED, \"connect\", \"\")")) connecting to /Users/olivier/Library/Containers/com.docker.docker/Data/s51
         com.docker.vpnkit is not running
         slirp check failed with: Unix.Unix_error(Unix.ECONNREFUSED, "connect", "")
[ERROR]  osxfs
         com.docker.osxfs is not running
[OK]     moby-console
[OK]     logs
[ERROR]  docker-cli
         Connection refused (ECONNREFUSED) connecting to /var/run/docker.sock: check if service is running
         Connection refused (ECONNREFUSED) connecting to /Users/olivier/Library/Containers/com.docker.docker/Data/s60: check if service is running
         docker ps failed
[OK]     menubar
[OK]     disk

I am running docker on a MacBook Pro (13-inch, 2016, Four Thunderbolt 3 Ports).

While being connected to a physical network, docker starts with no problem.
If connected to WIFI i see the above.

Any hints?

**Comment by tedmiston** [comment]
I'm experiencing this too on 1 of 2 machines (behind a VPN).

Possible dupe / related - https://github.com/docker/for-mac/issues/2502

**Comment by djs55** [comment]
It looks like the hypervisor crashed:
```
2018-01-22 21:52:56.212985+0100  localhost com.docker.driver.amd64-linux[57043]: rdmsr to register 0x64e on vcpu 0
2018-01-22 21:52:56.213038+0100  localhost com.docker.driver.amd64-linux[57043]: rdmsr to register 0x34 on vcpu 0
2018-01-22 21:52:56.323671+0100  localhost com.docker.driver.amd64-linux[57043]: com.docker.hyperkit: [INFO] block recycler: triggering background flush: 53423 used; 0 junk; 0 erased; 0 available; 0 copies; 0 roots; 0 Copying; 0 Copied; 0 Flushed; 0 Referenced; max_cluster = 53425
2018-01-22 21:52:58.396330+0100  localhost com.docker.driver.amd64-linux[57043]: vm exit[0]
2018-01-22 21:52:58.396408+0100  localhost com.docker.driver.amd64-linux[57043]: 	reason		VMX
2018-01-22 21:52:58.396520+0100  localhost com.docker.driver.amd64-linux[57043]: 	rip		0xffffffff8b57ebb8
2018-01-22 21:52:58.396578+0100  localhost com.docker.driver.amd64-linux[57043]: 	inst_length	3
2018-01-22 21:52:58.396661+0100  localhost com.docker.driver.amd64-linux[57043]: 	status		0
2018-01-22 21:52:58.396684+0100  localhost com.docker.driver.amd64-linux[57043]: virtio-net-vpnkit: interface will have uuid 4d702068-beae-4a05-a42f-7fa88313014d
2018-01-22 21:52:58.396712+0100  localhost com.docker.driver.amd64-linux[57043]: 	exit_reason	49
2018-01-22 21:52:58.396735+0100  localhost com.docker.driver.amd64-linux[57043]: virtio-net-vpnkit: Connection established with MAC=02:50:00:00:00:01 and MTU 1500
2018-01-22 21:52:58.396768+0100  localhost com.docker.driver.amd64-linux[57043]: 	qualification	0x0000000000000000
2018-01-22 21:52:58.396830+0100  localhost com.docker.driver.amd64-linux[57043]: 	inst_type		0
2018-01-22 21:52:58.396864+0100  localhost com.docker.driver.amd64-linux[57043]: 	inst_error		0
2018-01-22 21:52:58.396921+0100  localhost com.docker.driver.amd64-linux[57043]: COM1 connected to /dev/ttys000
2018-01-22 21:52:58.396982+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_PIN_BASED_CTLS:           0x000000000000003f
2018-01-22 21:52:58.397132+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_PRI_PROC_BASED_CTLS:      0x00000000b5186dfa
2018-01-22 21:52:58.397172+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_SEC_PROC_BASED_CTLS:      0x00000000000000aa
2018-01-22 21:52:58.397199+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_ENTRY_CTLS:               0x00000000000093ff
2018-01-22 21:52:58.397236+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_EXCEPTION_BITMAP:         0x0000000000040000
2018-01-22 21:52:58.397265+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_CR0_MASK:                 0x00000000e0000031
2018-01-22 21:52:58.397292+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_CR0_SHADOW:               0x0000000080050033
2018-01-22 21:52:58.397359+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_CR4_MASK:                 0x0000000000002000
2018-01-22 21:52:58.397431+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_CR4_SHADOW:               0x0000000000000000
2018-01-22 21:52:58.397460+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_PHYSICAL_ADDRESS:   0x00000000c0006110
2018-01-22 21:52:58.397493+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_LINEAR_ADDRESS:     0xffffaab240061110
2018-01-22 21:52:58.397517+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_CS_SELECTOR:        0x0000000000000010
2018-01-22 21:52:58.397647+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_CS_LIMIT:           0x00000000ffffffff
2018-01-22 21:52:58.397718+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_CS_ACCESS_RIGHTS:   0x000000000000a09b
2018-01-22 21:52:58.397761+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_CS_BASE:            0x0000000000000000
2018-01-22 21:52:58.397784+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_DS_SELECTOR:        0x0000000000000000
2018-01-22 21:52:58.397845+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_DS_LIMIT:           0x00000000ffffffff
2018-01-22 21:52:58.397883+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_DS_ACCESS_RIGHTS:   0x000000000001c000
2018-01-22 21:52:58.397914+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_DS_BASE:            0x0000000000000000
2018-01-22 21:52:58.397954+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_ES_SELECTOR:        0x0000000000000000
2018-01-22 21:52:58.397981+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_ES_LIMIT:           0x00000000ffffffff
2018-01-22 21:52:58.398020+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_ES_ACCESS_RIGHTS:   0x000000000001c000
2018-01-22 21:52:58.398050+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_ES_BASE:            0x0000000000000000
2018-01-22 21:52:58.398106+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_FS_SELECTOR:        0x0000000000000000
2018-01-22 21:52:58.398140+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_FS_LIMIT:           0x00000000ffffffff
2018-01-22 21:52:58.398171+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_FS_ACCESS_RIGHTS:   0x000000000001c000
2018-01-22 21:52:58.398222+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_FS_BASE:            0x0000000000000000
2018-01-22 21:52:58.398252+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_GS_SELECTOR:        0x0000000000000000
2018-01-22 21:52:58.398290+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_GS_LIMIT:           0x00000000ffffffff
2018-01-22 21:52:58.398316+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_GS_ACCESS_RIGHTS:   0x000000000001c000
2018-01-22 21:52:58.398436+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_GS_BASE:            0xffff98c1be600000
2018-01-22 21:52:58.398504+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_SS_SELECTOR:        0x0000000000000018
2018-01-22 21:52:58.398534+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_SS_LIMIT:           0x00000000ffffffff
2018-01-22 21:52:58.398571+0100  localhost com.docker.driver.amd64-linux[57043]: Hypervisor subprocess has shutdown
2018-01-22 21:52:58.398647+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_SS_ACCESS_RIGHTS:   0x000000000000c093
2018-01-22 21:52:58.398744+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_SS_BASE:            0x0000000000000000
2018-01-22 21:52:58.398790+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_LDTR_SELECTOR:      0x0000000000000000
2018-01-22 21:52:58.398819+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_LDTR_LIMIT:         0x00000000ffffffff
2018-01-22 21:52:58.398850+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_LDTR_ACCESS_RIGHTS: 0x000000000001c000
2018-01-22 21:52:58.398877+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_LDTR_BASE:          0x0000000000000000
2018-01-22 21:52:58.398907+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_TR_SELECTOR:        0x0000000000000040
2018-01-22 21:52:58.398930+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_TR_LIMIT:           0x0000000000002087
2018-01-22 21:52:58.398955+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_TR_ACCESS_RIGHTS:   0x000000000000008b
2018-01-22 21:52:58.398977+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_TR_BASE:            0xffff98c1be616100
2018-01-22 21:52:58.399003+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_GDTR_LIMIT:         0x000000000000007f
2018-01-22 21:52:58.399035+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_GDTR_BASE:          0xffff98c1be609000
2018-01-22 21:52:58.399065+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_IDTR_LIMIT:         0x00000000ffffffff
2018-01-22 21:52:58.399098+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_IDTR_BASE:          0x0000000000000000
2018-01-22 21:52:58.399129+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_CR0:                0x0000000080050033
2018-01-22 21:52:58.399155+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_CR3:                0x000000007b1ca000
2018-01-22 21:52:58.399181+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_CR4:                0x00000000000426b0
2018-01-22 21:52:58.399207+0100  localhost com.docker.driver.amd64-linux[57043]: VMCS_GUEST_IA32_EFER:          0x0000000000000d01
2018-01-22 21:52:58.399246+0100  localhost com.docker.driver.amd64-linux[57043]:
2018-01-22 21:52:58.399280+0100  localhost com.docker.driver.amd64-linux[57043]: rip: 0xffffffff8b57ebb8 rfl: 0x0000000000010046 cr2: 0x000000c420187008
2018-01-22 21:52:58.399337+0100  localhost com.docker.driver.amd64-linux[57043]: rax: 0xffffaab240061100 rbx: 0x0000000000000000 rcx: 0x00000000ffffff00 rdx: 0xffffaab240061000
2018-01-22 21:52:58.399407+0100  localhost com.docker.driver.amd64-linux[57043]: rsi: 0x0000000000000001 rdi: 0xffff98c1bee8c000 rbp: 0xffff98c1bee8c000 rsp: 0xffff98c1be603ec8
2018-01-22 21:52:58.406769+0100  localhost com.docker.supervisor[57040]: Reap com.docker.driver.amd64-linux (pid 57043): exit status 0
2018-01-22 21:52:59.409179+0100  localhost com.docker.supervisor[57040]: Stop 1 children with order 1: com.docker.driver.amd64-linux (pid 57043)
```
which is odd given that it shouldn't matter whether you're on the WiFi or not.

We've just released a new edge version which has enhanced diagnostics: could you give that a try and let me know how it goes? The download URL is https://download.docker.com/mac/edge/23360/Docker.dmg

Thanks for your report!

**Comment by dkd-dobberkau** [comment]
E9220915-6662-4227-B8EA-9458886CED8E

Docker for Mac: version: 18.03.0-ce-rc4-mac57 (dd2831d4b7421cf559a0881cc7a5fdebeb8c2b98)
macOS: version 10.12.6 (build: 16G1212)
logs: /tmp/E9220915-6662-4227-B8EA-9458886CED8E/20180322-182646.tar.gz
failure: com.docker.vpnkit is not running
[ERROR]  vpnkit
         Unexpected error ((Failure
  "Error connecting socket to 9p endpoint unix:/Users/olivier/Library/Containers/com.docker.docker/Data/s51: Unix.Unix_error(Unix.ECONNREFUSED, \"connect\", \"\")")) connecting to /Users/olivier/Library/Containers/com.docker.docker/Data/s51
         com.docker.vpnkit is not running
         vpnkit check failed with: Unix.Unix_error(Unix.ECONNREFUSED, "connect", "")
[OK]     vmnetd
[OK]     dns
[ERROR]  driver.amd64-linux
         com.docker.driver.amd64-linux is not running
[OK]     virtualization VT-X
[OK]     app
[OK]     moby
[OK]     system
[OK]     moby-syslog
[ERROR]  kubernetes
#ffb2b2#         kubernetes check failed with: (Failure
   "exec: /usr/local/bin/kubectl  --context docker-for-desktop --namespace docker get pods -o go-template='{{range .items}}{{.metadata.name}}\
  \n{{end}}': exit 1")##
[OK]     files
[OK]     env
[OK]     virtualization kern.hv_support
[ERROR]  osxfs
         com.docker.osxfs is not running
[OK]     moby-console
[OK]     logs
[ERROR]  docker-cli
         Connection refused (ECONNREFUSED) connecting to /var/run/docker.sock: check if service is running
         Connection refused (ECONNREFUSED) connecting to /Users/olivier/Library/Containers/com.docker.docker/Data/s60: check if service is running
         docker ps failed
[OK]     disk

**Comment by distracteddev** [comment]
I am also seeing this issue: 

Diagnostics ID: C2A3B06A-268C-46F6-B754-33EC2B523BE2

**Comment by toxuin** [comment]
I'm having similar issue here, and it happens quite often. Please have my diagnostic ID:

452F89C0-8878-4365-A84A-155AF5A33B2C

**Comment by roganartu** [comment]
tl;dr This appears to be caused by unavailable exported directories.

To reproduce go to `Preferences -> File Sharing` and add a directory that exists then `Apply & Restart`. Docker will start successfully. Quit Docker then delete the directory. Docker will fail to start with the error originally reported.

The vpnkit is a red herring, in all the diagnose outputs pasted above there's also:
```
[ERROR] osxfs
com.docker.osxfs is not running
```
I went looking through my docker logs and found the following:
```
2018-05-07 00:29:30.137782-0400  localhost com.docker.osxfs[19940]: Fatal unexpected exception: Unix.Unix_error(Unix.ENOENT, "readlink", "/mnt/home/tl")
2018-05-07 00:29:30.137944-0400  localhost com.docker.supervisor[19939]: com.docker.osxfs: stderr: osxfs: Unix.Unix_error(Unix.ENOENT, "readlink",
2018-05-07 00:29:30.137973-0400  localhost com.docker.supervisor[19939]: com.docker.osxfs: stderr:        "/mnt/home/tl")
2018-05-07 00:29:30.138843-0400  localhost com.docker.supervisor[19939]: cannot read stdout from com.docker.osxfs: EOF
2018-05-07 00:29:30.138863-0400  localhost com.docker.supervisor[19939]: cannot read stderr from com.docker.osxfs: EOF
2018-05-07 00:29:30.139543-0400  localhost com.docker.supervisor[19939]: Reap com.docker.osxfs (pid: 19940): exit status 1
2018-05-07 00:29:30.139854-0400  localhost com.docker.supervisor[19939]: Stop 1 children with order 1: [com.docker.driver.amd64-linux (pid: 19942)]
2018-05-07 00:29:30.139887-0400  localhost com.docker.supervisor[19939]: Send signal terminated to com.docker.driver.amd64-linux (pid: 19942)
2018-05-07 00:29:30.141159-0400  localhost com.docker.driver.amd64-linux[19942]: sending SIGTERM to com.docker.hyperkit pid: 19947
2018-05-07 00:29:30.155094-0400  localhost com.docker.vpnkit[19941]: Vmnet.Server.listen: read EOF so closing connection
2018-05-07 00:29:30.155106-0400  localhost com.docker.vpnkit[19941]: Vmnet.Server.listen returning Ok()
2018-05-07 00:29:30.155217-0400  localhost com.docker.driver.amd64-linux[19942]: Hypervisor subprocess has shutdown
2018-05-07 00:29:30.158636-0400  localhost com.docker.supervisor[19939]: cannot read stdout from com.docker.driver.amd64-linux: EOF
2018-05-07 00:29:30.158669-0400  localhost com.docker.supervisor[19939]: cannot read stderr from com.docker.driver.amd64-linux: EOF
2018-05-07 00:29:30.159382-0400  localhost com.docker.supervisor[19939]: Reap com.docker.driver.amd64-linux (pid: 19942): <nil>
2018-05-07 00:29:30.243252-0400  localhost com.docker.supervisor[19939]: Stop 2 children with order 2: [com.docker.osxfs (pid: 19940) com.docker.vpnkit (pid: 19941)]
2018-05-07 00:29:30.243292-0400  localhost com.docker.supervisor[19939]: Send signal terminated to com.docker.vpnkit (pid: 19941)
2018-05-07 00:29:30.245114-0400  localhost com.docker.supervisor[19939]: cannot read stdout from com.docker.vpnkit: EOF
2018-05-07 00:29:30.245177-0400  localhost com.docker.supervisor[19939]: cannot read stderr from com.docker.vpnkit: EOF
2018-05-07 00:29:30.245863-0400  localhost com.docker.supervisor[19939]: Reap com.docker.vpnkit (pid: 19941): signal: terminated
2018-05-07 00:29:30.345186-0400  localhost com.docker.supervisor[19939]: Supervisor caught an error
2018-05-07 00:29:30.345232-0400  localhost com.docker.supervisor[19939]: StopAll
2018-05-07 00:29:30.345257-0400  localhost com.docker.supervisor[19939]: Supervisor has failed, shutting down
```

Note that `osxfs` fails with `ENOENT` before `vpnkit` fails. If it is either removed from the exported directories by factory resetting, or made available again then Docker will start successfully. I suspect the wifi vs ethernet observation originally made was because there is an exported network directory available over @dkd-dobberkau's ethernet that isn't available over wifi.

So I guess `osxfs` needs to handle `ENOENT` a little more gracefully. I can't find the source for it (is it available?) otherwise I'd go digging further and submit a PR.

**Comment by YRM64** [comment]
Developers, if your working with edge or thinking about switching over, don't forget to include your new edge version with enhanced diagnostics for assistance; available for download at https://download.docker.com/Mac/edge/23360/Docker.dmg.

Thanks djs55 for your comments above.

**Comment by dkd-dobberkau** [comment]
@roganartu many thanks for the pointer here. Any idea on how I can see the shares bound to the interfaces?

**Comment by dkd-dobberkau** [comment]
Going back to Version 17.06.0-ce-mac18 (18433), Channel: stable, d9b66511e0 as this version works for me.
